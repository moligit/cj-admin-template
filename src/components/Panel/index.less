.wrapper {
  background-color: #fff;
  min-height: 100vh;
  margin: 20px;
  padding: 20px;
}
