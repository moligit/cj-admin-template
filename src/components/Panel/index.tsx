import React, { FC, useEffect } from 'react';
import { Spin } from 'antd';
import styles from './index.less';

interface ILayoutProps {
  children: React.ReactNode;
  loading?: boolean;
  title?: string;
}

const Panel: FC<ILayoutProps> = ({ children, title = '', loading = false }) => {
  useEffect(() => {
    document.title = title;
  }, []);

  return (
    <div className={styles.wrapper}>
      <Spin spinning={loading}>{children}</Spin>
    </div>
  );
};

export default Panel;
