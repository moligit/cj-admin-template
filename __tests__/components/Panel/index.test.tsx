// eslint-disable-next-line no-unused-vars
import React, { FC } from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Panel from '../../../src/components/Panel';

interface IDedecpProps {
  datetime: number;
}

const Dedecp: FC<IDedecpProps> = ({ datetime }) => (
  <p>
    <span>{datetime}</span>
    dede
  </p>
);
const dateobj = new Date();

describe('UI快照-panel组件', () => {
  test('renders correctly', () => {
    const wrapper = render(
      <Panel title="dedede">
        <Dedecp datetime={dateobj.getTime()} />
      </Panel>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
