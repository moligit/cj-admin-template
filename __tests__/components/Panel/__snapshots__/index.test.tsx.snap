// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI快照-panel组件 renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <p>
        <span>
          1614773272790
        </span>
        dede
      </p>
    </div>
  </div>
</div>
`;
